### PR TITLE
[SPARK-39859][SQL][FOLLOWUP] Only get ColStats when isExtended is true in Describe Column

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
get ColStats in `DescribeColumnExec` when `isExtended` is true

### Why are the changes needed?
To make code cleaner

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
existing test

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeColumnExec.scala
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeColumnExec.scala:51]: Looks like a clean move.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs) from my side.

We need @cloud-fan 's final sign-off because this is his request.

**Comment by huaxingao** [comment]
cc @cloud-fan Could you please take a look when you have a moment? Thanks!

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master/3.4!

**Comment by dongjoon-hyun** [comment]
Thank you, @huaxingao and @cloud-fan .

**Comment by huaxingao** [comment]
Thanks @cloud-fan @dongjoon-hyun

**Pull request closed without merging**
